### PR TITLE
Rename Callback.success() to onSuccess(), and failure() to onFailure().

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: retrofit/src/main/java/retrofit/DefaultCallAdapter.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+package retrofit;
+
+import java.lang.reflect.Type;
+
+/**
+ * A call adapter that uses the same thread for both I/O and application-level callbacks. For
+ * synchronous calls this is the application thread making the request; for asynchronous calls this
+ * is a thread provided by OkHttp's dispatcher.
+ */
+final class DefaultCallAdapter<T> implements CallAdapter<T> {
```
Comment (swankjesse) [inline retrofit/src/main/java/retrofit/DefaultCallAdapter.java]: Extracted from `NothingCallAdapter`

File: retrofit/src/main/java/retrofit/Callback.java [hunk 2]
```diff
@@ -30,8 +30,8 @@
  */
 public interface Callback<T> {
   /** Successful HTTP response. */
-  void success(Response<T> response);
+  void onSuccess(Response<T> response);
```
Comment (JakeWharton) [inline retrofit/src/main/java/retrofit/Callback.java]: onResponse to match OkHttp? Success might imply 2xx only and this gets called for all responses.

Comment (swankjesse) [inline retrofit/src/main/java/retrofit/Callback.java]: Can do.

**Comment by JakeWharton** [comment]
LGTM but needs rebased off master? The PR build (post-merge) is failing.

**Pull request merged**